File: elixir/darts.ex
defmodule Darts do
  @type position :: {number, number}
  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    dis = :math.sqrt(x * x + y * y)

    cond do
      dis > 10 -> 0
      dis > 5 -> 1
      dis > 1 -> 5
      true -> 10
    end
  end
end

File: dart/darts.dart
import 'dart:math';

class Darts {
  int score(double x, double y) {
    double distance = sqrt(x * x + y * y);
    if (distance <= 1) return 10;
    if (distance <= 5) return 5;
    if (distance <= 10) return 1;
    return 0;
  }
}
